### ISSUE TITLE
Feature request: Think about built in GDPR quick actions

### ISSUE THREAD

**Issue by zerosonesfun** [phpmyadmin/phpmyadmin#17130]
This is not related to a problem or existing issue. But, everyone talks about GDPR compliance these days. Usually, it's left to the actual website software to tackle. Not something like phpMyAdmin. But, what if?

What if phpMyAdmin added features that allowed website owners to, at the click of a button, download personal data? Every data structure is different, sure. But, I'm sure there would be a way to add a form/settings/buttons to help a website owner more quickly do certain data request tasks. Without having to build the features on the frontend if they don't want to.

The tech savvy professionals here will say, you can already do this. Type in a SQL query and bam. Done. But, correct me if I'm wrong here: One goal of phpMyAdmin is to provide a GUI. So, why not provide a GUI related to GDPR/privacy requests? Even if phpMyAdmin could only do so much (since every database has different tables), such as give the website owner a starting point... or help them like it helps you build queries... that would be better than nothing.

Think about it. Please. Don't just dismiss this.🤞

**Comment by ibennetch** [comment]
Hi @zerosonesfun and thanks for your suggestion. What do you propose? Since phpMyAdmin works in an abstract way with any data the user is storing, how could we make an interface that interacts with the many ways a user could store GDPR-related data? I'm interested to hear more about your ideas.

**Comment by zerosonesfun** [comment]
There could be a button/option/form above all tables (it would have to be near all tables because as you mentioned, the system doesn’t know which table houses user account data for example). If that option is activated, then phpMyAdmin knows, this table is for user accounts. Therefore, phpMyAdmin will magically know that it’s likely this table will need to be easily exported in different formats, or it will know that the webmaster may have to at any point quickly delete certain user data. Or anonymize data. And so it asks things like, which column is being used for the user identifier? You tell it. Now it lets you quickly search for and delete information throughout the database based on that ID without having to know SQL queries. 

I’m sure this would be a low priority for phpMyAdmin. But it was a thought I had the other day. Because sometimes you want to start a website using a certain open source software which may or may not have GDPR features and you may not be good with manually doing SQL queries. So if phpMyAdmin could help, that fills a gap. 

In short, in some way you tell phpMyAdmin: these are the tables, these are the columns, now help me with GDPR tasks. 😬
